app/src/org/commcare/android/adapters/IncompleteFormListAdapter.java

```diff
@@ -306,25 +334,41 @@ private void filterValues() {
 
         current.clear();
 
-        if (query == null || query.equals("")) {
+        if ("".equals(query)) {
```

Comment:
It was intentional: I made sure that 'query' was initialized to the empty string, so it should never be set to null, unless Editable.toString ever returns null, which I don't suspect it should.